DSharpPlus.Interactivity/EventHandling/Paginator.cs

```diff
@@ -138,16 +139,23 @@ async Task ResetReactionsAsync(IPaginationRequest p)
                 await msg.DeleteAllReactionsAsync("Pagination").ConfigureAwait(false);
             // ENDOF: 403 fix
 
-            if (emojis.SkipLeft != null)
-                await msg.CreateReactionAsync(emojis.SkipLeft).ConfigureAwait(false);
-            if (emojis.Left != null)
-                await msg.CreateReactionAsync(emojis.Left).ConfigureAwait(false);
-            if (emojis.Right != null)
-                await msg.CreateReactionAsync(emojis.Right).ConfigureAwait(false);
-            if (emojis.SkipRight != null)
-                await msg.CreateReactionAsync(emojis.SkipRight).ConfigureAwait(false);
-            if (emojis.Stop != null)
+            if (p.PageCount > 1)
+            {
+                if (emojis.SkipLeft != null)
+                    await msg.CreateReactionAsync(emojis.SkipLeft).ConfigureAwait(false);
+                if (emojis.Left != null)
+                    await msg.CreateReactionAsync(emojis.Left).ConfigureAwait(false);
+                if (emojis.Right != null)
+                    await msg.CreateReactionAsync(emojis.Right).ConfigureAwait(false);
+                if (emojis.SkipRight != null)
+                    await msg.CreateReactionAsync(emojis.SkipRight).ConfigureAwait(false);
+                if (emojis.Stop != null)
+                    await msg.CreateReactionAsync(emojis.Stop).ConfigureAwait(false);
+            }
+            else if (emojis.Stop != null && p is PaginationRequest paginationRequest && paginationRequest.PaginationDeletion == PaginationDeletion.DeleteMessage)
```

Comment:
I think checking for PaginationDeletion here with one page is inconsistent with multiple pages, so I would either remove this check or add this check to the `emojis.Stop` check where there are multiple pages.